Examples/Io/HepMC3/include/ActsExamples/Io/HepMC3/HepMC3Writer.hpp

```diff
@@ -1,33 +1,51 @@
 // This file is part of the Acts project.
 //
-// Copyright (C) 2018 CERN for the benefit of the Acts project
+// Copyright (C) 2018-2020 CERN for the benefit of the Acts project
 //
 // This Source Code Form is subject to the terms of the Mozilla Public
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 #pragma once
 
-#include <HepMC3/GenEvent.h>
+#include "ActsExamples/Framework/WriterT.hpp"
+#include "ActsExamples/Utilities/OptionsFwd.hpp"
+#include <string>
+#include <HepMC/GenEvent.h>
 #include <HepMC3/WriterAscii.h>
 
 namespace ActsExamples {
 
 /// HepMC3 event writer.
-struct HepMC3WriterAscii {
+class HepMC3AsciiWriter final : public WriterT<std::vector<HepMC3::GenEvent>> {
  public:
-  /// @brief Writes an event to file
-  /// @param writer writer of run files
-  /// @param event storage of the event
-  /// @return boolean indicator if the writing was successful
-  /// @note HepMC3 does not state a success or failure. The returned argument is
-  /// always true.
-  bool writeEvent(HepMC3::WriterAscii& writer,
-                  std::shared_ptr<HepMC3::GenEvent> event);
+  struct Config {
+    // The output directory
+    std::string outputDir;
+    // The stem of output file names
+    std::string outputStem;
+    // The input collection
+    std::string inputEvents;
+  };
 
-  /// @brief Reports the status of the writer
-  /// @param writer writer of run files
-  /// @return boolean status indicator
-  bool status(HepMC3::WriterAscii& writer);
+  /// @brief Constructor
+  ///
+  /// @param [in] cfg Config of the writer
+  /// @param [in] lvl The level of the logger
+  HepMC3AsciiWriter(const Config&& cfg, Acts::Logging::Level lvl);
+
+  /// @brief Writing method
```

Comment:
```suggestion
  /// Writing events to file.
```
Nitpicking: see above.

Reply:
done